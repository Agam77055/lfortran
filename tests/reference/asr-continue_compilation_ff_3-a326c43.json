{
    "basename": "asr-continue_compilation_ff_3-a326c43",
    "cmd": "lfortran --semantics-only --continue-compilation --no-color {infile}",
    "infile": "tests/errors/continue_compilation_ff_3.f",
    "infile_hash": "c0651cfa5989e8afd46b479ad40d6196bb8dbeaca048372296b3247c",
    "outfile": null,
    "outfile_hash": null,
    "stdout": null,
    "stdout_hash": null,
    "stderr": "asr-continue_compilation_ff_3-a326c43.stderr",
    "stderr_hash": "63d6b27aeac4f9e1c95c3f17b7a6e505266990235ad5ad628bef3e1c",
    "returncode": 1
}