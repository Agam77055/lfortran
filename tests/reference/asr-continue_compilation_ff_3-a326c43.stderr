syntax error: Token '::' is unexpected here
 --> tests/errors/continue_compilation_ff_3.f:9:9
  |
9 |     rea :: test_real(12)
  |         ^^ 

syntax error: Token ')' is unexpected here
  --> tests/errors/continue_compilation_ff_3.f:10:21
   |
10 |     real :: test_re()
   |                     ^ 

syntax error: The LFortran pragma !LF$ must be followed by a space
  --> tests/errors/continue_compilation_ff_3.f:99:9
   |
99 |         !LF$unroll 4  ! Error: Missing space after `!LF$`
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Module 'continue_compilation_3_fake_module' not declared in the current source and the modfile was not found
 --> tests/errors/continue_compilation_ff_3.f:2:5
  |
2 |     use continue_compilation_3_fake_module
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/continue_compilation_ff_3.f:19:9
   |
19 |     a = []
   |         ^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/continue_compilation_ff_3.f:21:16
   |
21 |     print *, [[[], [[]]], [[]], []]
   |                ^^ 

semantic error: Empty array constructor is not allowed
  --> tests/errors/continue_compilation_ff_3.f:22:16
   |
22 |     print *, [[[], [[]]], []]
   |                ^^ 

semantic error: Rank mismatch in array reference: the array `b` has rank `1`, but is referenced as rank `2`
  --> tests/errors/continue_compilation_ff_3.f:24:5
   |
24 |     b(:,:) = 1
   |     ^^^^^^ 

semantic error: Rank mismatch in array reference: the array `b` has rank `1`, but is referenced as rank `2`
  --> tests/errors/continue_compilation_ff_3.f:25:5
   |
25 |     b(:,:) = 2
   |     ^^^^^^ 

semantic error: Type member xx is not an array so it cannot be indexed.
  --> tests/errors/continue_compilation_ff_3.f:27:5
   |
27 |     y%xx(:) = 1
   |     ^^^^^^^ 

semantic error: Rank mismatch in array reference: the array `str` has rank `1`, but is referenced as rank `2`
  --> tests/errors/continue_compilation_ff_3.f:29:5
   |
29 |     str(1, 2)(:) = '1234'
   |     ^^^^^^^^^^^^ 

semantic error: Rank mismatch in array reference: the array `str` has rank `1`, but is referenced as rank `3`
  --> tests/errors/continue_compilation_ff_3.f:30:5
   |
30 |     str(1,2,3)(:) = '1234'
   |     ^^^^^^^^^^^^^ 

semantic error: Type-spec cannot contain an asterisk for a type parameter
  --> tests/errors/continue_compilation_ff_3.f:32:14
   |
32 |     print *, [character(*) :: "a", "b", "ball", "cat"]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Type-spec cannot contain an asterisk for a type parameter
  --> tests/errors/continue_compilation_ff_3.f:33:14
   |
33 |     print *, [character(*) :: "a2", "b2", "ball2", "cat2"]
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Different shape for array assignment on dimension 1(3 and 2)
  --> tests/errors/continue_compilation_ff_3.f:35:5
   |
35 |     x1 = reshape([1,2,3,4],[2,2])
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Different shape for array assignment on dimension 1(3 and 1)
  --> tests/errors/continue_compilation_ff_3.f:36:5
   |
36 |     x1 = reshape([1,2,3,4],[1,2])
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Arithmetic if (x) requires an integer or real for `x`
  --> tests/errors/continue_compilation_ff_3.f:40:9
   |
40 |     if ("yy") 1, 2, 3
   |         ^^^^ 

semantic error: Incorrect number of arguments passed to the 'size' intrinsic. It accepts at least 1 and at most 3 arguments.
  --> tests/errors/continue_compilation_ff_3.f:47:9
   |
47 |     i = size(a1, 1, 4, kind=4)
   |         ^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Incorrect number of arguments passed to the 'size' intrinsic. It accepts at least 1 and at most 3 arguments.
  --> tests/errors/continue_compilation_ff_3.f:48:9
   |
48 |     i = size()
   |         ^^^^^^ 

semantic error: Cannot assign to a constant variable
  --> tests/errors/continue_compilation_ff_3.f:50:5
   |
50 |     x3 = 1
   |     ^^^^^^ assignment here
   |
 7 |     integer, parameter :: x3 = 2
   |                           ~~~~~~ declared as constant

semantic error: The end variable of the data implied do loop must be constants
  --> tests/errors/continue_compilation_ff_3.f:53:22
   |
53 |     data(a1(i), i=1, k) / 1, 2, 3 /
   |                      ^ 

semantic error: The increment variable of the data implied do loop must be a constant
  --> tests/errors/continue_compilation_ff_3.f:55:25
   |
55 |     data(a1(i), i=1, 3, k) / 1, 2, 3 /
   |                         ^ 

semantic error: The start variable of the data implied do loop must be constants
  --> tests/errors/continue_compilation_ff_3.f:57:19
   |
57 |     data(a1(i), i=k, 3) / 1, 2, 3 /
   |                   ^ 

semantic error: Variable 'foo' is not declared
  --> tests/errors/continue_compilation_ff_3.f:60:9
   |
60 |     i = foo
   |         ^^^ 'foo' is undeclared

semantic error: Invalid argument `end` supplied
  --> tests/errors/continue_compilation_ff_3.f:62:5
   |
62 |     rewind(end="world")
   |     ^^^^^^^^^^^^^^^^^^^ 

semantic error: same_type_as is not implemented yet
  --> tests/errors/continue_compilation_ff_3.f:64:14
   |
64 |     print *, same_type_as(1.0, 1.0)
   |              ^^^^^^^^^^^^^^^^^^^^^^ 

semantic error: Kind of all the arguments of Sign must be the same
  --> tests/errors/continue_compilation_ff_3.f:66:14
   |
66 |     print *, sign(1, 1_8)
   |              ^^^^^^^^^^^^ 

semantic error: Argument 1 of dabs must be of double precision real type
  --> tests/errors/continue_compilation_ff_3.f:68:13
   |
68 |     print*, dabs(1)
   |             ^^^^^^^ 

semantic error: Argument of `sqrt` has a negative argument
  --> tests/errors/continue_compilation_ff_3.f:70:14
   |
70 |     print *, sqrt(-1.0)
   |              ^^^^^^^^^^ 

semantic error: Binary numeric operators cannot be used on strings
  --> tests/errors/continue_compilation_ff_3.f:72:14
   |
72 |     print *, "a" + "b"
   |              ^^^^^^^^^ help: use '//' for string concatenation

semantic error: The first index in string section is less than 1
  --> tests/errors/continue_compilation_ff_3.f:75:19
   |
75 |     print*, "s:", s(-1:4)
   |                   ^^^^^^^ 

semantic error: Substring `start` is less than one
  --> tests/errors/continue_compilation_ff_3.f:77:13
   |
77 |     print*, s1(-2:6)
   |             ^^^^^^^^ 

semantic error: Substring end index exceeds the string length
  --> tests/errors/continue_compilation_ff_3.f:80:13
   |
80 |     print*, s1(1: 9)
   |             ^^^^^^^^ 

semantic error: Substring end index at must be of type integer
  --> tests/errors/continue_compilation_ff_3.f:82:18
   |
82 |     print*, s1(1:5.2)
   |                  ^^^ 

semantic error: Substring start index at must be of type integer
  --> tests/errors/continue_compilation_ff_3.f:84:16
   |
84 |     print*, s1(1.1:5)
   |                ^^^ 

semantic error: Substring stride must be of type integer
  --> tests/errors/continue_compilation_ff_3.f:86:19
   |
86 |     print*, s(1:5:2.2)
   |                   ^^^ 

semantic error: Type mismatch in assignment, the types must be compatible
  --> tests/errors/continue_compilation_ff_3.f:88:5
   |
88 |     x = "x"
   |     ^   ^^^ type mismatch (integer and string)

semantic error: Type mismatch in binary operator, the types must be compatible
  --> tests/errors/continue_compilation_ff_3.f:90:9
   |
90 |     x = 5 + "x"
   |         ^   ^^^ type mismatch (integer and string)

semantic error: Type mismatch in assignment, the types must be compatible
  --> tests/errors/continue_compilation_ff_3.f:90:5
   |
90 |     x = 5 + "x"
   |     ^       ^^^ type mismatch (integer and string)

semantic error: Subroutine `bpe` called as a function
  --> tests/errors/continue_compilation_ff_3.f:93:9
   |
93 |     i = bpe()
   |         ^^^^^ 

semantic error: Variable 'xx' is not declared
  --> tests/errors/continue_compilation_ff_3.f:94:14
   |
94 |     print *, xx
   |              ^^ 'xx' is undeclared

semantic error: Variable 'test_re' is not declared
  --> tests/errors/continue_compilation_ff_3.f:95:5
   |
95 |     test_re = 1245.13
   |     ^^^^^^^ 'test_re' is undeclared

semantic error: Array reference is not allowed on scalar variable
  --> tests/errors/continue_compilation_ff_3.f:96:5
   |
96 |     c(1) = 1
   |     ^^^^ 

semantic error: Argument of `size` must be an array
   --> tests/errors/continue_compilation_ff_3.f:106:18
    |
106 |         print *, size(bpe)
    |                  ^^^^^^^^^ 

semantic error: Variable 'd' is not declared
   --> tests/errors/continue_compilation_ff_3.f:107:15
    |
107 |         bpe = d
    |               ^ 'd' is undeclared

semantic error: Assignment to subroutine is not allowed
   --> tests/errors/continue_compilation_ff_3.f:107:9
    |
107 |         bpe = d
    |         ^^^ 
